### PR TITLE
KYLIN-4091 support fast mode and simple mode for running CI

### PR DESCRIPTION
**Pull request by luguosheng1314**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: kylin-it/src/test/java/org/apache/kylin/provision/BuildCubeWithEngine.java [hunk 1]
```diff
@@ -398,15 +431,53 @@ private Boolean mergeSegment(String cubeName, long startDate, long endDate) thro
...
+        }
         ExecutableState state = waitForJob(job.getId());
         return Boolean.valueOf(ExecutableState.SUCCEED == state);
     }
+    private Boolean buildSegment(String cubeName, long startDate, long endDate) throws Exception {
+        return buildSegment(cubeName, startDate, endDate, false);
+    }
 
+    private Boolean checkJobState() throws Exception {
+        List<String> jobIds = jobService.getAllJobIds();
```
Comment (nichunen) [inline kylin-it/src/test/java/org/apache/kylin/provision/BuildCubeWithEngine.java]: why not use "getAllJobIdsInCache" to speed up?

File: kylin-it/src/test/java/org/apache/kylin/provision/BuildCubeWithEngine.java [hunk 2]
```diff
@@ -484,40 +555,56 @@ private void checkHFilesInHBase(CubeSegment segment) throws IOException {
...
 
     private CubeSegment getLastModifiedSegment(CubeInstance cube) {
-        return Collections.max(cube.getSegments(), new Comparator<CubeSegment>() {
-            @Override
-            public int compare(CubeSegment o1, CubeSegment o2) {
-                return Long.compare(o1.getLastBuildTime(), o2.getLastBuildTime());
-            }
-        });
+        Segments segments = cube.getSegments();
+        if (segments.size()>0) {
```
Comment (nichunen) [inline kylin-it/src/test/java/org/apache/kylin/provision/BuildCubeWithEngine.java]: Seems not formatted

File: kylin-it/src/test/java/org/apache/kylin/provision/BuildCubeWithEngine.java [hunk 3]
```diff
@@ -398,15 +431,53 @@ private Boolean mergeSegment(String cubeName, long startDate, long endDate) thro
...
+    }
 
+    private Boolean checkJobState() throws Exception {
+        List<String> jobIds = jobService.getAllJobIds();
+        while (true) {
+            if (jobIds.size() == 0) {
+                return true;
+            }
+            for (int i = jobIds.size() - 1; i >= 0; i--) {
+                AbstractExecutable job = jobService.getJob(jobIds.get(i));
```
Comment (nichunen) [inline kylin-it/src/test/java/org/apache/kylin/provision/BuildCubeWithEngine.java]: use "getOutputDigest" is better

**Pull request merged**